src/pybamm/models/full_battery_models/base_battery_model.py

```diff
@@ -990,6 +990,16 @@ def options(self, extra_options):
                 f"must use surface formulation to solve {self!s} with hydrolysis"
             )
 
+        if isinstance(self, pybamm.lithium_ion.DFN):
+            if options["working electrode"] == "positive":
+                if (
+                    options["thermal"] == "lumped"
```

Comment:
Should this be something like checking if thermal is defined at all for the half cell? This would make me think that non-lumped models still work

Reply:
Other thermal models are disabled in the options checking (https://github.com/pybamm-team/PyBaMM/blob/0e0936a2ce27b9ce908958a33f79895f149ea232/src/pybamm/models/full_battery_models/base_battery_model.py#L608). This would be pretty unwieldy to do here because I am trying to only disable the DFN case, which I don't think I can check in the options checker.